bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -172,9 +174,37 @@ public boolean tableExists(String tableName) throws IOException {
     return response;
   }
 
+  // BigtableTableAdmin#listTables doesn't include table details (like column families), so the
+  // table descriptors must be fetched into 2 phases:
+  // 1. list all of the names
+  // 2. fetch the table details
+  //
+  // Due to the non-atomic listing, tables in the list might disappear before fetching the details.
+  // This method will suppress those tables.
+  private HTableDescriptor[] getTableDescriptorsIgnoreFailure(TableName[] tableNames)
+      throws IOException {
+    List<HTableDescriptor> descriptors = new ArrayList<>();
+    for (TableName tableName : tableNames) {
+      try {
+        descriptors.add(getTableDescriptor(tableName));
+      } catch (IOException ex) {
+        // This will suppress TableNotFoundException and FailedPreconditionException as both of
+        // these can occur due to race condition.
+        if (!(ex instanceof TableNotFoundException
```

Comment:
Please add a comment that TableNotFoundException comes from the Hbase layer and the FailedPrecondition originates from the bigtable client layer.